libraries/Microsoft.Bot.Builder.Dialogs/Prompts/PromptCultureModels.cs

```diff
@@ -235,28 +235,27 @@ public static class PromptCultureModels
         /// <returns>Normalized locale.</returns>
         public static string MapToNearestLanguage(string cultureCode)
         {
-            cultureCode = cultureCode.ToLowerInvariant();
-
-            if (SupportedLocales.All(o => o != cultureCode))
+            if (!string.IsNullOrWhiteSpace(cultureCode))
             {
-                // Handle cases like EnglishOthers with cultureCode "en-*"
-                var fallbackCultureCodes = SupportedLocales
-                    .Where(o => o.EndsWith("*", StringComparison.Ordinal) &&
-                                cultureCode.StartsWith(o.Split('-').First(), StringComparison.Ordinal)).ToList();
+                cultureCode = cultureCode.ToLowerInvariant();
 
-                if (fallbackCultureCodes.Count == 1)
+                if (!SupportedLocales.Contains(cultureCode))
                 {
-                    return fallbackCultureCodes.First();
-                }
+                    var culturePrefix = cultureCode.Split('-').First();
+                    var fallbackLocales = SupportedLocales.Where(locale => locale.StartsWith(culturePrefix, StringComparison.Ordinal));
 
```

Comment:
Multiple enumerations of fallbackLocales in the lines that follow. Consider converting to List.